### PR TITLE
How to implement object oriented programming Matlab

### LINKED ISSUE #3007: How to implement object oriented programming using Matlab
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Introduction:
Object-oriented programming is a design approach that enables you to programmatically define structures called objects that combine data (properties) together with functions that operate on that data (methods). In Matlab, you can create objects that model the behavior of devices and systems in the real world. Those objects can then be used as building blocks in applications used to simulate and analyze complex systems.
Using object-oriented programming in Matlab, you can manage software complexity by organizing your code into logical components that are easier to maintain and extend. You can avoid code duplication by creating reusable objects with well-defined interfaces that hide the complexity of the underlying code. Furthermore, your objects can evolve and change over time without introducing incompatibilities in client code. In this tutorial, we will look at how to apply this in Matlab

### Key takeaways: 
- understanding object oriented programming.
- Importance of object oriented programming.
- How to apply this skills in Matlab.
### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by collince777**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**

This closes #3007.      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by marienjus** [comment]
@hectorkambow, if this article is not a duplicate then I'm willing to review it.

**Comment by marienjus** [comment]
Hello @collince777 
I will take you through the review beginning with preliminary revision where I will check your article for grammar accuracy and more.

**Comment by marienjus** [comment]
Please add hero image. Also, it seems like this is your first contribution and therefore, you need to add author details @collince777.

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/how-to-implement-object-oriented-programming-in-matlab/index.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+SS =  Salary(objS,10)   %calculates the salary from sales
+```
+The value 10 is the number of hours. When we run the codes separately, we get the salary and the sales.
+
+![image showing results](/engineering-education/how-to-implement-object-oriented-programming/oop_five.png)
+The calculated salary is different as per the mathematical operations.
+
+### Abstraction
+It can be said to be an extension of encapsulation. To implement it in our example, we can only define the function named `salary` in the base class employed, and it's derived will have the function implementation.
```
Comment (marienjus) [inline content/articles/how-to-implement-object-oriented-programming-in-matlab/index.md:162]: Hey seems you've used "It" a lot. Kindly rewrite this in every instance of the OOP introduction to improve readability.

**Comment by marienjus** [comment]
@collince777 ping me when done so that I can proceed with the tech review. I've checked through the article twice to ensure that everything is well. :rocket:

**Comment by collince777** [comment]
@marienjus requested changes made

**Comment by marienjus** [comment]
Great! I will now proceed to a tech review.
cc: @ahmadmardeni1 @hectorkambow

**Comment by marienjus** [comment]
> @marienjus what is the progress of this article?

I like to do tech review twice. Today I'm checking for code best practices and it will be ready for final review. Kindly be patient. I've done massive progress. I actually had communicated with @hectorkambow that I'm having this stuff as part of my units this semester. 🔥

**Comment by marienjus** [comment]
I love the way you walk the reader through the concepts. Checked that everything is okay and this is a great one!
Ready to ping content moderators for a final review :+1:

**Comment by WanjaMIKE** [comment]
@srishilesh I am doing a final review. No need for commits on this PR. Thanks

**Comment by hectorkambow** [comment]
Overall nice job.  @collince777 

Please be sure to go my final edits (and the article again) to improve future articles and reviews. The tutorial was a bit difficult and rough to follow along in certain parts. 

cc @marienjus

**Pull request merged**